broadlink/climate.py

```diff
@@ -201,3 +201,252 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
+            data['mode'] = 'D'
+        elif mode == 0x80:
+            data['mode'] = 'H'
+        elif mode == 0xc0:
+            data['mode'] = 'F'
+        else:
+            data['mode'] = 'unrecognized value'
+
+        speed_L = payload[0x0f]
+        speed_R = payload[0x10] 
+        if speed_L == 0x60 and speed_R == 0x00:
+            data['speed'] = 'L'
+        elif speed_L == 0x40 and speed_R == 0x00:
+            data['speed'] = 'M'
+        elif speed_L == 0x20 and speed_R == 0x00:
+            data['speed'] = 'H'
+        elif speed_L == 0x40 and speed_R == 0x80:
+            data['speed'] = 'Mu'
+        elif speed_R == 0x40:
+            data['speed'] = 'T'
+        elif speed_L == 0xa0 and speed_R == 0x00:
+            data['speed'] = 'A'
+        else:
+            data['speed'] = 'unrecognized value'
+
+        data['sleep'] = True if (payload[0x1a] & 0b100 == 0b000) else False
+        data['display'] = (payload[0x16] & 0x10 == 0x10)
+        data['health'] = (payload[0x14] & 0b11 == 0b11)
+
+        checksum = self._calculate_checksum(payload[:0x19]) # checksum=(payload[0x1a] << 8) + payload[0x19]
+
+        if (payload[0x19] == checksum[0] and payload[0x1a] == checksum[1]):
+            pass # success
+        else:
+            print('checksum fail', ['{:02x}'.format(x) for x in checksum])
+
+        if (payload):
+            data['received_payload'] = payload
+
+        return data
+
+    def get_ac_info(self, payload:bool=None) -> dict:
+        """Returns dictionary with A/C info...
+        Not implemented yet, except power state.
+
+        Args:
+            payload (Optional[bool]): add the received payload for debugging
+        """
+        payload = self._send_short_payload(0)
+
+        # first 13 bytes are the same: 22 00 bb 00 07 00 00 00 18 00 01 21 c0
+        data = {}
+        data['state'] = True if (payload[0x0d] & 0b1 == 0b1) else False
+
+        if (payload):
+            data['received_payload'] = payload
+
+        return data
+
+    def set_advanced(self, state:bool=None, mode:str=None, set_temp:float=None, speed:str=None, swing_v:str=None, swing_h:str=None,
+                        sleep:bool=None, display:bool=None, health:bool=None) -> bytes:
+        """Set paramaters of unit and return response.
+        If not all parameters are specificed, will try to derive from the unit's current state. Doesn't work when powered down.
+        Args:
+            state (bool): power
+            set_temp (float): temperature set point 16<n<32
+            mode (str): Cooling, Heating, Fan, Dry, Auto (use first letter)
+            speed (str): Mute, Low, Mid, High, Turbo (available only in cooling) (use first letter and Mu for Mute)
+            swing_h (str): ON, OFF
+            swing_v (str): ON, OFF, 1, 2, 3, 4, 5 (fixed positions)
+            sleep (bool): 
+            display (bool):
+            health (bool):
+        """
+
+        args = locals()
+        args.pop('self')
```

Comment:
Please do not use locals(), it makes the code difficult to maintain.

Reply:
Is [this](https://github.com/Enosh/python-broadlink/blob/d885cc5d133c94f4201c6a2ce990021149a199a4/broadlink/climate.py#L393) a good way to do this?